stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/composer/MessageComposerView.kt

```diff
@@ -65,31 +40,125 @@ public class MessageComposerView : ConstraintLayout {
     private fun init() {
         binding = StreamUiMessageComposerBinding.inflate(streamThemeInflater, this)
         binding.leadingContent.apply {
+            val defaultLeadingContent = DefaultLeadingContent(context)
             removeAllViews()
-            addView(leadingContent(this))
+            addView(defaultLeadingContent)
         }
         binding.centerContent.apply {
+            val defaultCenterContent = DefaultCenterContent(context).apply {
+                onTextChangedListener = { onInputChanged(it) }
+                onClearButtonClickListener = { onClearInput() }
+            }
             removeAllViews()
-            addView(centerContent(this))
+            addView(defaultCenterContent)
         }
         binding.trailingContent.apply {
+            val defaultTrailingContent = DefaultTrailingContent(context).apply {
+                this.onSendButtonClickListener = { onSendMessage() }
+            }
             removeAllViews()
-            addView(trailingContent(this))
+            addView(defaultTrailingContent)
+        }
+    }
+
+    public fun renderState(state: MessageInputState) {
+        (binding.trailingContent.children.first() as? DefaultTrailingContent)?.renderState(state)
+        (binding.centerContent.children.first() as? DefaultCenterContent)?.renderState(state)
+        (binding.leadingContent.children.first() as? DefaultLeadingContent)?.renderState(state)
+    }
+}
+
+internal class DefaultLeadingContent : FrameLayout {
```

Comment:
Should these  be inner classes, but public maybe? What if our users want to use them but customize the behavior a bit?

Reply:
Yes. This makes sense to me